splink/estimate_u.py

```diff
@@ -51,23 +51,38 @@ def estimate_u_values(linker: "Linker", target_rows):
         for cl in cc.comparison_levels:
             cl._level_dict["tf_adjustment_column"] = None
 
-    sql = """
-    select count(*) as count
-    from __splink__df_concat_with_tf
-    """
-    dataframe = training_linker._sql_to_splink_dataframe_checking_cache(
-        sql, "__splink__df_concat_count"
-    )
-    result = dataframe.as_record_dict()
-    dataframe.drop_table_from_database()
-    count_rows = result[0]["count"]
-
     if settings_obj._link_type in ["dedupe_only", "link_and_dedupe"]:
+        sql = """
```

Comment:
You could also edit the logic for `link_and_dedupe` here too, as that will have the same issues you've identified with the link_only job.

Reply:
Ah no, you're correct on this one.

I was under the impression that we only deduped the first dataset for some reason, but this looks to be incorrect.